x-pack/platform/plugins/shared/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -759,7 +748,7 @@ export class AlertsClient<
     const params: Record<string, string[]> = {};
 
     idsToUpdate.forEach((id) => {
-      const newAlert = newAlerts.find((alert) => alert.getUuid() === id);
+      const newAlert = newAlerts.find((alert) => alert.getUuid() === id || alert.getId() === id);
```

Comment:
detection alerts generate their own alert UUID which they use to index the alert document and they use this UUID as the alert ID as well. The UUID that's auto-generated for detection alerts has no meaning. We have this [issue](https://github.com/elastic/kibana/issues/144862) open to allow rule executors to pass in their pre-generated UUID but until then, we will expand the comparison to match against alert UUID or alert ID.

Reply:
Updated in https://github.com/elastic/kibana/pull/232307/commits/0ca068cca463c7dca2e7f594e8651ddc2e75a5d4